### PR TITLE
[RFC] Added single checkbox function

### PR DESCRIPTION
**Pull request by abhisheksavant**
@jskulski @alexindigo

### CODE REVIEW

File: test/test-checkbox.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    this.stub(this.testObject, '_trigger');
+    // return event type instead of full object
+    this.stub(this.testObject, '_createEvent').returnsArg(0);
+
+    // invoke test subject
+    this.testObject.checkbox(target, function()
+    {
+      // each event type triggered
+      // callback ran within testObject context
+      assert.calledWith(this._trigger, target, 'touchstart');
```
Comment (alexindigo) [inline test/test-checkbox.js]: To make it true to the unit test spirint, can you stub `_triggerEvents` and check what's passed to it?
Similar to other tests, like `test-type.js`.

Comment (alexindigo) [inline test/test-checkbox.js]: Sorry, confused myself.

File: test/test-checkbox.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+buster.testCase('checkbox',
+{
+  setUp: common.setUp,
+
+  'Invokes _triggerEvents with list of events': function(done)
+  {
+    var eventsList = ['touchstart', ['touchend', 'mouseover', 'mousedown', 'focus', 'focusin', 'DOMFocusIn', 'mouseup', '[Function: selectCheckbox]', 'change', 'click'], ['mouseout', 'blur', 'focusout', 'DOMFocusOut']]
+      , target     = common.createTargetElement.call(this)
+      , callback = this.spy()
```
Comment (alexindigo) [inline test/test-checkbox.js:16]: Can you clean up formatting? Everything else looks good. Merge at will. Thank you.

**Comment by alexindigo** [comment]
:+1:

File: test/test-checkbox.js (hunk 2 above)
Comment (asavant-trulia) [inline test/test-checkbox.js:16]: +1

**Comment by asavant-trulia** [comment]
I dont think I have merge power, @alexindigo can you merge this PR, all the checks are done.

**Comment by alexindigo** [comment]
Because you didn't fix formatting. :) jk.
You're not you. :) @asavant-trulia isn't collaborator on this project and @abhisheksavant is. :)

**Pull request merged**